apps/ledger-live-mobile/scripts/merge-timing-files.sh

```diff
@@ -0,0 +1,204 @@
+existing_file="$ARTIFACTS_DIR/e2e-test-results-$PLATFORM.json"
+output_file="./e2e-test-results-$PLATFORM.json"
+
+# Check if shard files exist
+if ! find . -maxdepth 1 -name "$pattern" -print -quit | grep -q .; then
+  log "No timing files found for pattern $pattern, skipping merge."
+  if [ -f "$existing_file" ]; then
+    log "Keeping existing cached timing file"
+    cp "$existing_file" "$output_file"
+  else
+    log "Creating empty timing file"
+    echo '{}' > "$output_file"
+  fi
+  exit 0
+fi
+
+# List all shard files found
+shard_files=($pattern)
+log "Found ${#shard_files[@]} shard files: ${shard_files[*]}"
+
+# Validate and filter files
+valid_files=()
+invalid_files=()
+
+for file in "${shard_files[@]}"; do
+  log "Validating file: $file"
+  
+  # Check if file exists and is not empty
+  if [ ! -s "$file" ]; then
+    log "  ✗ File is empty or doesn't exist: $file"
+    invalid_files+=("$file")
+    continue
+  fi
+  
+  # Check if file contains valid JSON
+  if ! jq empty "$file" 2>/dev/null; then
+    log "  ✗ Invalid JSON in file: $file"
+    invalid_files+=("$file")
+    continue
+  fi
+  
+  # Check if file is a JSON object (not null, array, string, etc.)
+  json_type=$(jq -r 'type' "$file" 2>/dev/null)
+  if [ "$json_type" != "object" ]; then
+    log "  ✗ File is not a JSON object (type: $json_type): $file"
+    invalid_files+=("$file")
+    continue
+  fi
+  
+  # Check if file has testResults field
+  if ! jq -e '.testResults' "$file" >/dev/null 2>&1; then
+    log "  ⚠ File missing testResults field: $file"
+    # Still consider it valid but warn
+  fi
+  
+  log "  ✓ Valid file: $file"
+  valid_files+=("$file")
+done
+
+log "Valid files: ${#valid_files[@]}"
+if [ ${#invalid_files[@]} -gt 0 ]; then
+  log "ERROR: Found ${#invalid_files[@]} invalid files that cannot be processed:"
+  for file in "${invalid_files[@]}"; do
+    log "  - $file"
+  done
+  log "Failing merge due to invalid files. Please investigate and fix these files."
+  exit 1
+fi
+
+# Perform merge based on whether existing file exists
+if [ -f "$existing_file" ]; then
+  log "Found existing cached timing file, performing incremental merge..."
+  
+  if [ ${#valid_files[@]} -gt 0 ]; then
+    log "Merging ${#valid_files[@]} valid files with existing cache..."
+    
+    # Incremental merge with existing file
+    jq -s '
+      # Get existing and new test results with null checks
+      (.[0].testResults // []) as $existing |
+      # Filter out null testResults and add valid ones
+      ([.[1:][] | select(.testResults != null) | .testResults] | add // []) as $new |
+      # Create a map of existing tests by name for easy lookup
+      ($existing | map({key: .name, value: .}) | from_entries) as $existingMap |
+      # Create a map of new tests by name
+      ($new | map({key: .name, value: .}) | from_entries) as $newMap |
+      # Merge maps: new tests override existing ones
+      ($existingMap + $newMap) as $mergedMap |
```

Comment:
Ask: So for each run we are overwriting the old values ? What about failing test taking more/less time than usual and how are handled retries ?